addons/uom/models/uom_uom.py

```diff
@@ -4,20 +4,70 @@
 from odoo import api, fields, tools, models, _
 from odoo.exceptions import UserError, ValidationError
 
+def _check_category_reference_uniqueness(cat):
+    cnt=sum(uom.uom_type=='reference' and uom.active for uom in cat.uom_ids)
```

Comment:
reference_count = sum...

pep8 + we usually prefer to use explicit variable name than short abbreviation.

Reply:
Why is this outside the class?

Reply:
As the function is used in classes UoMCategory & UoM

Reply:
Since your only argument is a category then you should add it in uom.category class. You replace cat by self. And when you need to call it from uom then you call uom.category_id._check_category_reference_uniqueness()

Doing function outside of class is not a good practice and should only be used in particular cases.